src/scripts/node_detector_control.py

```diff
@@ -43,46 +66,73 @@ def receive_img(self, msg: image):
             self.pub.publish(
                 observations(msg.camera_id, msg.image_counter,
                              observation_list))
 
     def __init__(self):
-        self.input_sub = rospy.Subscriber("camera/images", image,
-                                          self.receive_img)
+        # Locked when self.detectors is altered or iterated
+        self.detect_lock = threading.Lock()
+
+        ## Topic with images to analyze
+        self.input_sub = rospy.Subscriber(
+            '{}/{}'.format(name_node_camera, topic_images), image,
+            self.receive_img)
+
         ## Topic to publish results
-        self.pub = rospy.Publisher("{}/observations".format(rospy.get_name()),
+        self.pub = rospy.Publisher('{}/{}'.format(rospy.get_name(),
+                                                  topic_observations),
                                    observations,
                                    queue_size=50)
 
-        rospy.Service("{}/add_object_detector".format(rospy.get_name()),
-                      srv.add_object_detector, self.add_object_detector)
+        rospy.Service(
+            '{}/{}'.format(rospy.get_name(), srv_add_object_detector),
+            srv.add_object_detector, self.add_object_detector)
 
-        rospy.Service("{}/remove_object_detector".format(rospy.get_name()),
+        rospy.Service('{}/{}'.format(rospy.get_name(), srv_rm_object_detector),
                       srv.remove_object_detector, self.remove_object_detector)
 
-        # Combine results to single message, or publish separately.
-        self.combine = rospy.get_param("combine_results", True)
-        rospy.Service("{}/combine_toggle".format(rospy.get_name()), srv.toggle,
-                      self.toggle_combine)
+        ## Load all required rosparams 
+        self.load_rosparams()
 
+        self.detect_lock.acquire()
         # temporary, TODO: replace with parameter based ?
         for kp in rospy.get_param("testi", {}).items():
             # TODO: model and label _path should be renamed to _file
-            model_path = '{}/{}'.format(tflite_path, kp[1]['model_path'])
-            label_path = '{}/{}'.format(tflite_path, kp[1]['label_path'])
+            # TODO: If we actually need the parameter expansion, then we need to
+            #       map the model_path and label_path items in the dict.
             self.detectors[kp[0]] = ObjectNode(name=kp[0],
-                                               model_path=model_path,
-                                               label_path=label_path)
+                                               model_path=kp[1]['model_path'],
+                                               label_path=kp[1]['label_path'],
+                                               frequency=kp[1]['frequency'],
+                                               detect_on=kp[1]['detect_on'],
+                                               score_threshold=kp[1]['score_threshold']
+            )
```

Comment:
Miksi nämä on kovakoodattu? Osa niistä, esim score_threshold eivät ole pakollisia, joten niitä ei välttämättä ole.

Reply:
Alunperin sen takia, että jos käyttää parameter expansionia, niin koodiin jää helposti bugeja jos muuttaa jotain olemassaolevaa koodia. Näin kuten se nyt on, niin näkee selkeästi mitä argumentteja annetaan. Kysyinkin äsken discordissa kun tätä koodasin, että halutaanko käyttää alkuperäistä **kp[1] muotoa (onko meillä syytä käyttää sitä), vai asetetaanko kaikki selkeästi? Varmaan et tätä kuullut, kun en saanut vastausta. Mutta toki tuo on hyvä pointti, että voidaan antaa vain tiettyjä argumentteja helposti, jos käytetään parameter expansiona. Muutan tuon takaisin sitten.

Reply:
Ja siis aivan alunperin tuo tuli, kun piti lisätä tflite_path noihin. Mutta nyt tuo on korjattu pullarissa